daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/AssertPatternParsers.scala

```diff
@@ -75,7 +76,9 @@ class AssertPatternParser(
       val isMatch = dis.lookingAt(m, start)
       if (!isMatch) {
         val message = getAssertFailureMessage(start)
-        val diag = new AssertionFailed(context.schemaFileLocation, start, message)
+        val currentElem = start.infoset
+        val details = "\nParsed value was: " + currentElem.toString
+        val diag = new AssertionFailed(context.schemaFileLocation, start, message, Some(details))
```

Comment:
Instead of Some(details), which is scala's Option type, which requires allocation of an object, can you use One(details), a Maybe type, which doesn't allocate? 

For this to work and not allocate it can't be a varargs list at the end of the AssertionFailed constructor. Has to be a specific argument of that type.

Reply:
Will do, I simply forgot about the One() function.